core/src/banking_stage/forward_packet_batches_by_accounts.rs

```diff
@@ -333,4 +322,122 @@ mod tests {
             assert_eq!(2, batches.next().unwrap().len());
         }
     }
+
+    #[test]
+    fn test_get_batch_index_by_updated_costs() {
+        let test_cost = 99;
+
+        // check against vote limit only
+        {
+            let mut forward_packet_batches_by_accounts =
+                ForwardPacketBatchesByAccounts::new_with_default_batch_limits();
+            forward_packet_batches_by_accounts.batch_vote_limit = test_cost + 1;
+
+            let transaction_cost = TransactionCost::SimpleVote {
+                writable_accounts: vec![],
+            };
+            assert_eq!(
+                0,
+                forward_packet_batches_by_accounts.get_batch_index_by_updated_costs(
+                    &transaction_cost,
+                    &UpdatedCosts {
+                        updated_block_cost: test_cost,
+                        updated_costliest_account_cost: 0
+                    }
+                )
+            );
+            assert_eq!(
+                1,
+                forward_packet_batches_by_accounts.get_batch_index_by_updated_costs(
+                    &transaction_cost,
+                    &UpdatedCosts {
+                        updated_block_cost: test_cost * 2,
```

Comment:
nit: this should probably be test_cost + 1 to test the boundary condition?

Reply:
100%, updated!